library/Notifications/Widget/ItemList/IncidentHistoryListItem.php

```diff
@@ -85,55 +83,46 @@ protected function assembleHeader(BaseHtmlElement $header): void
         $header->add(new TimeAgo($this->item->time->getTimestamp()));
     }
 
-    protected function getIncidentEventIcon(): string
+    protected function getIconName(): string
     {
         switch ($this->item->type) {
             case 'opened':
             case 'incident_severity_changed':
-                return $this->getSeverityIcon();
-            case 'recipient_role_changed':
-                return $this->getRoleIcon();
+                switch ($this->item->new_severity) {
+                    case 'ok':
+                        return Icons::OK;
+                    case 'err':
+                        return Icons::ERROR;
+                    case 'crit':
+                        return Icons::CRITICAL;
+                    default:
+                        return Icons::WARNING;
+                }
             case 'closed':
                 return Icons::CLOSED;
+            case 'recipient_role_changed':
+                switch ($this->item->new_recipient_role) {
+                    case 'manager':
+                        return Icons::MANAGE;
+                    case 'subscriber':
+                        return Icons::SUBSCRIBED;
+                    default:
+                        if ($this->item->old_recipient_role !== null) {
+                            if ($this->item->old_recipient_role === 'manager') {
+                                return Icons::UNMANAGE;
+                            } else {
+                                return Icons::UNSUBSCRIBED;
+                            }
+                        }
+                        return '';
+                }
             case 'rule_matched':
                 return Icons::RULE_MATCHED;
+            case 'notified':
+                return Icons::NOTIFIED;
             case 'escalation_triggered':
                 return Icons::TRIGGERED;
             default:
-                return Icons::NOTIFIED;
-        }
-    }
-
-    protected function getSeverityIcon(): string
-    {
-        switch ($this->item->new_severity) {
-            case 'ok':
-                return Icons::OK;
-            case 'err':
-                return Icons::ERROR;
-            case 'crit':
-                return Icons::CRITICAL;
-            default:
-                return Icons::WARNING;
-        }
-    }
-
-    protected function getRoleIcon(): ?string
-    {
-        switch ($this->item->new_recipient_role) {
-            case 'manager':
-                return Icons::MANAGE;
-            case 'subscriber':
-                return Icons::SUBSCRIBED;
-            default:
-                if ($this->item->old_recipient_role !== null) {
-                    if ($this->item->old_recipient_role === 'manager') {
-                        return Icons::UNMANAGE;
-                    } else {
-                        return Icons::UNSUBSCRIBED;
-                    }
-                }
-
                 return '';
```

Comment:
Now we're getting somewhere. It's now very obvious that the removal of `getSeverityIcon` and `getRoleIcon` is only of cosmetic nature. You even introduced a styling error in the `recipient_role_changed` switch case as there's a missing empty line before the `default`'s return statement, which was there before.

So, to me, a typical case of refactorization which has no meaningful advantage. The only required change in this method is that by default no icon is shown. Please reset the method so that this is the only remaining change.